### PR TITLE
STM F3 support

### PR DESCRIPTION
**Pull request by eziosoft**
Partial F3 support. Should be enough to flash correctly.
Currently only full chip erase is supported.

IT IS NOT TESTED AT ALL as I don't have hardware

### CODE REVIEW

File: js/protocols/stm32.js [hunk 1]
```diff
@@ -342,6 +344,8 @@ STM32_protocol.prototype.verify_flash = function (first_array, second_array) {
 STM32_protocol.prototype.upload_procedure = function (step) {
     var self = this;
 
+    var extendedErase=false;
```
Comment (hydra) [inline js/protocols/stm32.js]: this and the other references to 'extendedErase' needed to be `self.extendedErase`.

**Comment by hydra** [comment]
Merged into development, see 90cc08c86bfd5e0add12f065a78fd07dc045b55a

**Comment by hydra** [comment]
Thanks for this contribution!

**Comment by eziosoft** [comment]
I'm curious if it works with F3. Can you confirm that ?

**Comment by hydra** [comment]
Yes, it works!  

I also added support for the extended erase for only-needed pages.

See 5471ab7f652384ea703a7658dfa7cb18faffc6a6

However I found that using the default baud rate if 9xxxxx never worked, 4xxxxx worked occasionally 256000 and below works every time so I did this:

3086a9882800b1aaa615af249007af5e4afe9bb1

**Comment by hydra** [comment]
f3 flashing

**Comment by hydra** [comment]
``
SERIAL: Connection opened with ID: 186, Baud: 115200
stm32.js:90 Sending ascii "R" to reboot
serial.js:135 SERIAL: Connection with ID: 186 closed, Sent: 1 bytes, Received: 0 bytes
serial.js:83 SERIAL: Connection opened with ID: 187, Baud: 256000
stm32.js:355 STM32 - Serial interface initialized on the MCU side
stm32.js:390 STM32 - Bootloader version: 3.1
stm32.js:406 STM32 - Signature: 0x422
stm32.js:304 Chip recognized as F3 STM32F30xxx, STM32F31xxx
stm32.js:442 Executing local erase (via extended erase)
stm32.js:475 Erasing. pages: 0x00 - 0x36, checksum: 0x34
stm32.js:479 Erasing: done
stm32.js:543 Writing data ...
stm32.js:605 Writing: done
stm32.js:618 Verifying data ...
stm32.js:336 Verification successful, matching: 65536 bytes
stm32.js:336 Verification successful, matching: 43296 bytes
stm32.js:690 Programming: SUCCESSFUL
stm32.js:720 Sending GO command: 0x8000000
serial.js:135 SERIAL: Connection with ID: 187 closed, Sent: 116626 bytes, Received: 111413 bytes
stm32.js:754 Script finished after: 16.945 seconds
```

**Pull request closed without merging**